web/src/hooks/useСheckPublicName.ts

```diff
@@ -0,0 +1,45 @@
+import { useDispatch } from 'react-redux';
+import useMutation from 'use-mutation';
+import { p2pUrl } from 'web/src/api/config';
+import { fetchJson, FetchError } from 'web/src/helpers/fetch';
+import { alertPush } from 'web/src/modules/public/alert/actions';
+
+interface CheckPublicNameInput {
+  publicName?: string;
+}
+
+const сheckPublicName = async (params: CheckPublicNameInput) => {
+  const response = await fetchJson(`${p2pUrl()}/profile/check-public-name`, {
+    method: 'POST',
+    body: JSON.stringify(params),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+    credentials: 'include',
+  });
+
+  return response;
+};
+
+export const useCheckPublicName = () => {
+  const dispatch = useDispatch();
+  const [mutate, data] = useMutation<CheckPublicNameInput, any, unknown>(сheckPublicName, {
```

Comment:
```suggestion
  return useMutation<CheckPublicNameInput, any, unknown>(сheckPublicName, {
```
Можно сразу возвращать результат